pydra/engine/core.py

```diff
@@ -166,12 +167,16 @@ def checksum(self):
         """
         input_hash = self.inputs.hash
         if self.state is None:
-            self._checksum = create_checksum(self.__class__.__name__, input_hash)
+            self._checksum = create_checksum(
+                self.__class__.__name__, self.name, input_hash
```

Comment:
why should the name change the checksum? name does not carry any compute information, which are in inputs, workflow graph/function.

Reply:
It would be nice to make it easily available to users which directories pertain to which task, but you are right that it doesn't carry compute information. 

I'll revert for now but open up an issue regarding this.

Reply:
the cache will no longer behave like current nipype. this mapping exists and is exposed with `output_dir` and `result` at the task/workflow level. @djarecka - has tests for this.

Reply:
yes, `output_dir` should contain all directories with results

Reply:
i'm checking `output_dir` in all/most workflow tests. If `self.state` than the `output_dir` contains multiple paths